### PR TITLE
Fix ipi-conf-azure-oidc-creds-provision release image issue

### PR DESCRIPTION
**Pull request by huangmingxia**
[CCO-234](https://issues.redhat.com/browse/CCO-234)
[OCPQE-15990](https://issues.redhat.com/browse/OCPQE-15990)
[OCPQE-15999](https://issues.redhat.com/browse/OCPQE-15999)

Using clusterbot to create Azure Workload Identity cluster failed. The reason for the failure is  `unauthorized: authentication required` could not extract azure credentials requests from the release image. 
```
error: unable to read image registry.ci.openshift.org/ocp/release:4.14.0-0.nightly-2023-07-18-085740: unauthorized: authentication required
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"k8s.io/test-infra/prow/entrypoint/run.go:79","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2023-07-19T06:44:12Z"}
error: failed to execute wrapped command: exit status 1 
```
The clusterbot log here: https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-launch-azure-modern/1681554755987443712

### CODE REVIEW

**Comment by huangmingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-workload-identity-tp-p1-f14

**Comment by huangmingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-workload-identity-tp-p1-f14

**Comment by huangmingxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-azure-ipi-workload-identity-tp-p1-f14

File: ci-operator/step-registry/ipi/conf/azure/oidc-creds-provision/ipi-conf-azure-oidc-creds-provision-ref.yaml [hunk 1]
```diff
@@ -7,6 +7,9 @@ ref:
     requests:
       cpu: 10m
       memory: 100Mi
+  dependencies:
+    - name: "release:latest"
+      env: RELEASE_IMAGE_LATEST_FROM_BUILD_FARM
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/oidc-creds-provision/ipi-conf-azure-oidc-creds-provision-ref.yaml:12]: I am curious where did you reference the workaround?

Comment (huangmingxia) [reply ci-operator/step-registry/ipi/conf/azure/oidc-creds-provision/ipi-conf-azure-oidc-creds-provision-ref.yaml:12]: @jianlinliu When I saw this problem with azure, I created a cluster with [gcp Workload Identity using clusterbot](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/release-openshift-origin-installer-launch-gcp-modern/1681562927326302208) and didn't notice the problem.  So I went to reference aws and gcp and realized it was an image issue. 

For the workaround, I found it here: https://github.com/search?q=repo%3Aopenshift%2Frelease%20RELEASE_IMAGE_LATEST_FROM_BUILD_FARM&type=code and then I took a trip to learn how to use it.

**Pull request merged**